include/nuttx/sched.h

```diff
@@ -529,6 +530,12 @@ struct task_group_s
   tls_ndxset_t tg_tlsset;           /* Set of TLS data indexes allocated        */
 #endif
 
+  /* Task-specific Data *********************************************************/
+
+#ifndef CONFIG_BUILD_KERNEL
+  FAR struct libvars_s *tg_libvars; /* C library global variables */
+#endif
+
```

Comment:
This field is for convenience only.  By buffering a copy of the allocated per-task variables, it saves having to look up the pointer when a pthread inherits it.  No essential but a good simplifier.